File: 00-每日一题/20200223_350.cpp
#include <vector>
#include <stdlib.h>
#include <iostream>
#include <map>
#include <unordered_map>
#include <set>
#include <queue>
#include <algorithm>
using namespace std;


class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> myMap;
        vector<int> res;
        for(int x:nums1)
        {
            myMap[x]+=1;
        }
        for(int x:nums2){
            if(myMap[x]>0)
            {
                res.push_back(x);
                myMap[x]--;
            }
        }
        return res;
    }
};

// 如果两个数组都是排好序的 应该怎么做
class Solution2 {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        int i=0,j=0;
        while(i<nums1.size()&&j<nums2.size())
        {
            if(nums1[i]==nums2[j])
            {
                res.push_back(nums1[i]);
                i++;
                j++;
            }
            else if(nums1[i]<nums2[j])
            {
                i++;
            }
            else
                j++;
        }
        return res;
    }
};

int main()
{
    Solution s;
    vector<int> a = {1,2,4,5,7,100};
    vector<int> b = {0,2,2,98};
    vector<int> c = s.intersect(a,b);
    for(int i=0;i<c.size();i++)
        cout << c[i] << endl;
}

File: 00-每日一题/20200223_350.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        count = {}
        for n in nums1:
            count[n] = count.get(n, 0) + 1

        result = []
        for n in nums2:
            if n in count and count[n] > 0:
                result.append(n)
            count[n] = count.get(n, 0) - 1
        return result

if __name__ == '__main__':
    obj = Solution()
    while True:
        nums_str = input().strip().split()
        nums = list(map(int, nums_str))
        nums_str2 = input().strip().split()
        nums2 = list(map(int, nums_str2))

        res = obj.intersect(nums, nums2)
        print(res)
